course5/homework6-try2/mystr.cc

```diff
@@ -0,0 +1,130 @@
+#include "mystr.h"
+#include <memory>
+#include <cmath> 
+#include <cstring>
+
+using namespace mystring;
```

Comment:
It is better to simply wrap the content of the file in a namespace. Otherwise you can get weird linking issues with free functions.